From 2014f6d9934fb065bc7f02cf7c78fadc2e98415f Mon Sep 17 00:00:00 2001
From: Lokesh Vutla <lokeshvutla@ti.com>
Date: Wed, 21 Oct 2015 13:47:42 +0530
Subject: [PATCH 603/800] drivers: net: davinci_cpdma: Fix build warning with
 LPAE enabled

Following build warning appears if LPAE is enabled:

In file included from include/linux/dma-mapping.h:86:0,
                 from drivers/net/ethernet/ti/davinci_cpdma.c:21:
		 drivers/net/ethernet/ti/davinci_cpdma.c: In function 'cpdma_desc_pool_create':
		 drivers/net/ethernet/ti/davinci_cpdma.c:182:48: warning: passing argument 3 of 'dma_alloc_attrs' from incompatible pointer type
		    pool->cpumap = dma_alloc_coherent(dev, size, &pool->phys,
		                                                    ^

The phys vaiable in cpdma_desc_pool structure is only used to allocate dma,
so it should be of type dma_addr_t instead of phys_addr_t.

Fixes: c767db5147f0f ("net: eth: davinci_cpdma: 64-bit phys/dma_addr_t cleanup")
Signed-off-by: Lokesh Vutla <lokeshvutla@ti.com>
[zou: Original patch taken from PROCESSOR-SDK-LINUX-AM335X 02_00_01_07]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/net/ethernet/ti/davinci_cpdma.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/ti/davinci_cpdma.c b/drivers/net/ethernet/ti/davinci_cpdma.c
index 657b65b..d23df64 100644
--- a/drivers/net/ethernet/ti/davinci_cpdma.c
+++ b/drivers/net/ethernet/ti/davinci_cpdma.c
@@ -81,7 +81,7 @@ struct cpdma_desc {
 };
 
 struct cpdma_desc_pool {
-	phys_addr_t		phys;
+	dma_addr_t		phys;
 	u32			hw_addr;
 	void __iomem		*iomap;		/* ioremap map */
 	void			*cpumap;	/* dma_alloc map */
-- 
1.7.5.4

